pyuvdata/uvdata/mir_parser.py

```diff
@@ -10,1082 +10,6383 @@
+                return np.array(
+                    [self._header_key_index_dict[key] for key in header_key], dtype=int
                 )
-                # This bit of code is to trap an unfortunately
-                # common problem with metadata of MIR autos not
-                # being correctly recorded.
-                if data_offset == 0:
-                    if (file_size % (4 * (2**14) * nchunks * 2 + 20)) != 0:
-                        nchunks = int(auto_vals["nChunks"][0])
-                        if (file_size % (4 * (2**14) * nchunks * 2 + 20)) != 0:
-                            raise IndexError(
-                                "Could not determine auto-correlation record size!"
-                            )
-                    # How big the record is for each data set
-                    last_offset = 4 * (2**14) * int(nchunks) * 2
-                    ac_data = np.zeros(
-                        file_size // ((4 * (2**14) * int(nchunks) * 2 + 20)),
-                        dtype=ac_read_dtype,
-                    )
-                ac_data[marker] = (
-                    auto_vals["scan"][0],
-                    marker + 1,
-                    auto_vals["antenna"][0],
-                    nchunks,
-                    last_offset + 20,
-                    data_offset,
-                    auto_vals["dhrs"][0],
+
+        # At this point, we expect to hand back a boolean mask, so either instantiate
+        # it or make a copy of the supplied mask argument.
+        mask = (
+            self._mask.copy() if use_mask else np.full(len(self), bool(and_where_args))
+        )
+
+        if where is not None:
+            # Otherwise, if we are going through where statements, then use the where
+            # method to build a mask that we can use to select the data on. Check to
+            # make sure that where matches what we expect - want to both accept a tuple
+            # and sequence of tuples, so force it to be the latter.
+            try:
+                if not (isinstance(where[0], (tuple, list))):
+                    # If where is not indexable, it'll raise a TypeError here.
+                    where = [where]
+                for item in where:
+                    # Note we raise a TypeError in this loop to trap an identical bug,
+                    # namely that the user has not provided a valid argument for where.
+                    if len(item) != 3:
+                        raise TypeError
+            except TypeError:
+                raise ValueError(
+                    "Argument for where must be either a 3-element tuple, or sequence "
+                    "of 3-element tuples."
                 )
-                data_offset += last_offset + 20
-                marker += 1
-        return ac_data
 
-    @staticmethod
-    def parse_vis_data(filepath, in_start_dict, sp_data):
+            # Now actually start going through the where statements.
+            where_success = False
```

Comment:
On review, I actually realized this if statement was superfluous, so I've just stripped it out to prevent `where_success` from being accessed before being assigned a value. Changed in 514d9922.